lib/runtime/life/exports.go

```diff
@@ -141,12 +140,6 @@ func (in *Instance) ExecuteBlock(block *types.Block) ([]byte, error) {
 
 	// remove seal digest only
 	for _, d := range block.Header.Digest.Types {
-		// hack since substrate node_runtime can't seem to handle BABE pre-runtime digests
-		// with type prefix (ie Primary, Secondary...)
-		if bytes.Equal(in.version.SpecName(), []byte("node")) {
-			break
-		}
```

Comment:
Do we still need this?

If so, should we then keep it and change it to query the version from the runtime and do this if condition check?

Reply:
The unit test currently seems to indicate so, probably because the `gssmr` runtime is not secondary slot aware.

Reply:
I agree, I'm currently working on updating the host API and dev/gssmr runtimes to be compatible with polkadot v0.9.17. if we merge this, it will temporarily break the unit test but we can just skip it until the runtime is updated

Reply:
Which test should we skip? The `t.Skip("reason")` should be part of this PR.

Reply:
`TestInstance_ExecuteBlock_GossamerRuntime`